src/test/java/com/microsoft/sqlserver/jdbc/unit/statement/PreparedStatementTest.java

```diff
@@ -0,0 +1,202 @@
+        assertSame(SQLServerConnection.getDefaultPrepareStatementOnFirstCall(), conn1.getPrepareStatementOnFirstCall());        
+        conn1.setPreparedStatementDiscardActionThreshold(SQLServerConnection.getDefaultPreparedStatementDiscardActionThreshold() + 1);
+        conn1.setPrepareStatementOnFirstCall(!SQLServerConnection.getDefaultPrepareStatementOnFirstCall());
+        assertNotSame(SQLServerConnection.getDefaultPreparedStatementDiscardActionThreshold(), conn1.getPreparedStatementDiscardActionThreshold());        
+        assertNotSame(SQLServerConnection.getDefaultPrepareStatementOnFirstCall(), conn1.getPrepareStatementOnFirstCall());        
+        
+        // Verify new instance not same as changed instance.
+        SQLServerConnection conn2 = (SQLServerConnection)DriverManager.getConnection(connectionString);
+        assertNotSame(conn1.getPreparedStatementDiscardActionThreshold(), conn2.getPreparedStatementDiscardActionThreshold());        
+        assertNotSame(conn1.getPrepareStatementOnFirstCall(), conn2.getPrepareStatementOnFirstCall());        
+
+        // Verify instance setting is followed.
+        SQLServerConnection.setDefaultPreparedStatementDiscardActionThreshold(SQLServerConnection.INITIAL_DEFAULT_PREPARED_STATEMENT_CLEANUP_THRESHOLD);
+        try (SQLServerConnection con = (SQLServerConnection)DriverManager.getConnection(connectionString)) {
+            try {
+
+                String query = "/*unprepSettingsTest*/SELECT * FROM sys.objects;";
+
+                // Verify initial default is not serial:
+                assertTrue(1 < SQLServerConnection.getDefaultPreparedStatementDiscardActionThreshold());
+
+                // Verify first use is batched.
+                try (SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement)con.prepareStatement(query)) {
+                    pstmt.execute();
+                }
+                // Verify that the un-prepare action was handled immediately.
+                assertSame(1, con.outstandingPreparedStatementDiscardActionCount());
+
+                // Force un-prepares.
+                con.forcePreparedStatementDiscardActions();
+
+                // Verify that queue is now empty.
+                assertSame(0, con.outstandingPreparedStatementDiscardActionCount());
+
+                // Set instance setting to serial execution of un-prepare actions.
+                con.setPreparedStatementDiscardActionThreshold(1);                
+
+                try (SQLServerPreparedStatement pstmt = (SQLServerPreparedStatement)con.prepareStatement(query)) {
+                    pstmt.execute();
+                }
+                // Verify that the un-prepare action was handled immediately.
+                assertSame(0, con.outstandingPreparedStatementDiscardActionCount());
+            } 
+            finally {
```

Comment:
BTW, the reason for fun is just because it is easy to comment in/out a catch-block.